task16.py

```diff
@@ -0,0 +1,19 @@
+"""  Требуется вычислить, сколько раз встречается некоторое число X в массиве A[1..N]. Пользователь в первой строке вводит натуральное число N – количество элементов в массиве. В последующих  строках записаны N целых чисел Ai. Последняя строка содержит число X
+
+*Пример:*
+
+5
+    1 2 3 4 5
+    3
+    -> 1
+    """
+import random
+N = int(input('Введите натуральное число:'))
+lst = [random.randrange(10) for _ in range(N)]
+print(f'{lst = }')
+k = int(input('Введите число k:'))
+count = 0
+for i in range(len(lst)):
+    if lst[i] == k:
```

Comment:
Во-первых, совершенно не обязательно считать длину списка. У вас уже имеется переменная, которая содержит этот параметр. Во-вторых, вообще не нужно использовать функцию `range`, достаточно просто пройтись по элементам списка
```suggestion
for i in st:
    if i == k:
```